actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -332,6 +283,51 @@ def escape_once(html)
         ERB::Util.html_escape_once(html)
       end
 
+      # Returns a string of attributes built from +options+.
+      #
+      # ==== Escape
+      #
+      # The generated attributes are escaped by default. This can be disabled using
+      # +escape+.
+      #
+      #   tag_options(src: 'open & shut.png')
+      #   # => " src=\"open &amp; shut.png\""
+      #
+      #   tag_options(src: 'open & shut.png', true)
+      #   # => " src=\"open & shut.png\""
+      #
+      # ==== Examples
+      #
+      #   tag_options(class: "button", disabled: true)
+      #    # => " class=\"button\" disabled=\"disabled\""
+      #
+      #   tag_options(data: {user_id: "1"})
+      #    # => " data-user-id=\"1\""
+      def tag_options(options, escape = true)
```

Comment:
Don't think we've done our due diligence of combing over the implementation to ready it as public API. The leading space and `escape` not being a keyword argument are two cases of this.

I feel uneasy about going with this, but I can't quite place my finger on what would change that right now.

Reply:
@kaspth I agree that the leading space should probably be removed 👍 . Looking at the use cases in this file, it should be an easy change to make.

The `escape = true` API matches `tag` (https://github.com/rails/rails/pull/39105/files#diff-3d863df5229eeafadb6df53160605a5bR199) and `content_tag` (https://github.com/rails/rails/pull/39105/files#diff-3d863df5229eeafadb6df53160605a5bR235) in this file, so I think it makes sense to use the same API there.

If there's anything that made me feel uncomfortable while working on this, it was the name of this method. I think `tag_attributes` would be more appropriate as it returns HTML attributes.

Having this helper will enable developers to make use of some of the complexity managed by `tag` and `content_tag` without leaning on Rails to build an entire HTML element. I'm hoping we can come up with a solution that gives us this benefit ❤️ 

What do you think?

Reply:
Thanks for the feedback. As @joelhawksley has outlined, the decisions here are mostly trying to minimize changes to existing call sites of this helper.

I took a stab at removing the leading space, which feels awkward to me as then the caller will need to decide if it needs a leading space, but it doesn't necessarily know if `tag_options` would return anything.

Reply:
:wave:  I added 2cdca0f which includes the `tag_attributes` rename and the leading space removal.